kieker-monitoring/src/kieker/monitoring/writer/raw/AbstractRawDataWriter.java

```diff
@@ -0,0 +1,38 @@
+/***************************************************************************
+ * Copyright 2015 Kieker Project (http://kieker-monitoring.net)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+package kieker.monitoring.writer.raw;
+
+import kieker.common.configuration.Configuration;
+
+/**
+ * Abstract class for raw data writers.
+ *
+ * @author Holger Knoche
+ *
+ * @since 1.13
+ */
+public abstract class AbstractRawDataWriter implements IRawDataWriter {
+
+	/**
+	 * Creates a new instance using the given configuration
+	 * @param configuration The configuration to use
+	 */
+	public AbstractRawDataWriter(final Configuration configuration) { // NOPMD
```

Comment:
Basically for two reasons. One is ease of extension in case we decide to introduce abstract functionality which needs the configuration. The second is "Poka Yoke": The concrete writer has to provide such a constructor anyway, so it's (IMHO) nicer to enforce it right away than to let the implementor fail at runtime.